### PR TITLE
Clarify single node cluster syntax

### PR DESCRIPTION
**Pull request by m1nkeh**
This clarifies the issue I found here: #525

### CODE REVIEW

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -77,15 +77,46 @@ resource "databricks_cluster" "shared_autoscaling" {
...
+  node_type_id            = data.databricks_node_type.smallest.id
+  autotermination_minutes = 20
+
+  spark_conf = {
+    # Single-node
+    "spark.databricks.cluster.profile" : "singleNode"
+    "spark.master" : "local[*]"
+  }
+
+  custom_tags = {
```
Comment (alexott) [inline docs/resources/cluster.md:115]: same as above

**Review by alexott — CHANGES_REQUESTED** [review]

File: docs/resources/cluster.md [hunk 2]
```diff
@@ -77,15 +77,46 @@ resource "databricks_cluster" "shared_autoscaling" {
...
+
+* `min_workers` - (Optional) The minimum number of workers to which the cluster can scale down when underutilized. It is also the initial number of workers the cluster will have after creation.
+* `max_workers` - (Optional) The maximum number of workers to which the cluster can scale up when overloaded. max_workers must be strictly greater than min_workers.
+
+When using a [Single Node cluster](https://docs.databricks.com/clusters/single-node.html), `num_workers` needs to be `0`. It can be set to `0` explicitly, or simply not specified, as it defaults to `0`.  When `num_workers` is `0`, provider checks for presence of the required Spark configurations:
 * `spark.master` must has prefix `local`, like `local[*]`
 * `spark.databricks.cluster.profile` must have value `singleNode`
 
+and also `custom_tag` entry:
+* `"ResourceClass" = "SingleNode"`
```
[the remark this answers is not in the document]
Comment (m1nkeh) [reply docs/resources/cluster.md:90]: nope, it does not work - I was running it last night!

You end up with a `standard` cluster

File: docs/resources/cluster.md (hunk 1 above)
Comment (m1nkeh) [reply docs/resources/cluster.md:115]: no, you need the custom tag

File: docs/resources/cluster.md [hunk 3]
```diff
@@ -77,15 +77,46 @@ resource "databricks_cluster" "shared_autoscaling" {
...
+```hcl
+data "databricks_node_type" "smallest" {
+  local_disk = true
+}
+
+data "databricks_spark_version" "latest_lts" {
+  long_term_support = true
+}
+
+resource "databricks_cluster" "shared_autoscaling" {
```
[the remark this answers is not in the document]
Comment (m1nkeh) [reply docs/resources/cluster.md:103]: nice catch

File: docs/resources/cluster.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexott) [reply docs/resources/cluster.md:90]: Ok, that's for the AWS only... not required on Azure - just checked there. 
thank you for raising that - I'll pass that to the docs team & engineering...

[the remark this answers is not in the document]
Comment (m1nkeh) [reply docs/resources/cluster.md:90]: you comment confuses me? are you suggesting that this `"ResourceClass = "SingleNode"` is not required on Azure? As you have just checked on Azure?

I only use Azure (never even logged in to AWS), and can confirm that my investigations last night that this was required - it was most bizarre.

**Comment by nfx** [comment]
https://github.com/databrickslabs/terraform-provider-databricks/blob/master/compute/acceptance/cluster_test.go#L67 let's see what this integration test shows today.

File: docs/resources/cluster.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexott) [reply docs/resources/cluster.md:90]: for me it works just fine on Azure, and you don't need to have azurerm provider for that.  It looks like that there are some problems in API...

**Review by alexott — APPROVED** [review]
now it looks good to me, thank you for raising this up

**Pull request merged**